include/mata/utils/closed-set.hh

```diff
@@ -361,9 +361,11 @@ ClosedSet<T> ClosedSet<T>::intersection(const ClosedSet<T>& rhs) const {
     // Iterates through all the tuples from Antichan1 X Antichan2
     // and creates an union of them
     if(type_ == ClosedSetType::upward_closed_set) {
-        for(auto element1 : antichain_) {
-            for(auto element2 : rhs.antichain()) {
-               result.insert(element1.Union(element2));
+        for(Node element1 : antichain_) {
```

Comment:
use `const Node&` instead? (to avoid copying)

Reply:
aha, I see, ok. 
(only that all of this is going to get removed in the next pull request. )